### DISCUSSION TITLE
Unable to Properly Uninstall Application after Slipstreaming Patch in Bundle

### DISCUSSION THREAD

**Discussion by KarthikChandran303** [wixtoolset/issues#8760] [Questions]
Hi all, 
I have a Bundler that includes a "MainInstaller" msi package and a patch (which is slipstreamed) within a Chain like this:

`<Chain>`
  `<MsiPackage Id="MainInstaller" SourceFile="$(var.MainInstaller)" bal:DisplayInternalUICondition="WixBundleAction >= 0" Visible="no" Vital="yes" Permanent="no"/>`
  `<MspPackage Id="Patch" SourceFile="$(var.Patch)" Vital="yes" After="MainInstaller" Slipstream="yes" Permanent="no"/>`
`</Chain>`

Uninstalling the bundle only removes the slipstreamed patch.  Any ideas as to why the main installer is not being removed?

I am also providing a part of my log which gets generated when I uninstall:

> Planned package: Patch, state: Present, default requested: Absent, ba requested: Absent, execute: Uninstall, rollback: Install, default cache strategy: Keep, ba requested strategy: Keep, cache: NonVital, uncache: Yes, dependency: Unregister, expected install registration state: Absent, expected cache registration state: Absent
> [1B80:1844][2024-09-19T16:24:19]i218:    Plan 1 patch targets for package: Patch
> [1B80:1844][2024-09-19T16:24:19]i219:    Planned patch target: {GUID}, state: Present, default requested: Absent, ba requested: Absent, execute: Uninstall, rollback: Install
> [1B80:1844][2024-09-19T16:24:19]i201: Planned package: MainInstaller, state: Superseded, default requested: Absent, ba requested: Absent, execute: None, rollback: None, default cache strategy: Keep, ba requested strategy: Keep, cache: No, uncache: No, dependency: Unregister, expected install registration state: Ignored, expected cache registration state: Ignored
> [1B80:1844][2024-09-19T16:24:19]i220:    Plan 1 slipstream patches for package: MainInstaller
> [1B80:1844][2024-09-19T16:24:19]i221:       Planned slipstreamed patch: Patch, execute: None, rollback: None
> [1B80:1844][2024-09-19T16:24:19]i299: Plan complete, result: 0x0

Thanks for the help!

**Comment by ListechAu** [comment]
Hi @KarthikChandran303,

I'm having a similar problem. I started looking at migrating my installation code from WiX 3.x to v6.0.0 and have found that WiX 6.0.0 generated bundles only uninstall the patch whereas in WiX 3.x both the MSI and the slipstreamed MSP were uninstalled when the bundle was uninstalled.

Did you ever find a solution to this issue?

Thanks, Darren

**Reply by wmanning** [reply]
Maybe related to this: https://github.com/wixtoolset/issues/issues/7778?

**Reply by ListechAu** [reply]
Thanks for your reply and I apologise for the delayed response, but I have been away on annual leave.

In the case of #7778 their problem relates to multiple bundles with mixed versions of WiX Toolset, whereas my problem relates to a single v6.0.1 bundle that contains an msi and an msp that is slipstreamed during installation of the bundle. However, when the bundle is uninstalled only the slipstreamed msp is being uninstalled and the baseline files from the msi remain installed.    

The bundle code snippet is something like that below:

`<Chain>`
`	<MsiPackage Id="UpdateService_x64.msi" SourceFile="$(var.BaseMsiFolder)UpdateService_x64.msi" />`
`	<MspPackage Id="UpdateService_x64.msp" SourceFile="$(var.CurrMspFolder)UpdateService_x64.msp" Slipstream="yes" />`
`</Chain>`

I would like to move our WiX v3 code over to v6 but this is causing me problems so any help would be greatly appreciated.

Thanks, Darren

**Reply by kalyani1217** [reply]
@ListechAu , I'm having the same issue.How did you fix the issue?

**Reply by ListechAu** [reply]
Hi @kalyani1217,  

Unfortunately, I have not been able to solve this issue so I'm still using WiX v3 code :-(

Regards, Darren

**Reply by robmen** [reply]
Hmm, is there an issue tracking this problem? WiX v3 is out of community service, so that is not a safe place to sit. I highly encourage taking a proactive stance to help track down the problem.

There are tests for [Burn slipstreaming](https://github.com/wixtoolset/wix/blob/main/src/test/burn/WixToolsetTest.BurnE2E/SlipstreamTests.cs). Is there a case slipping through the cracks? Adding a test to reproduce the problem would be a very good first step towards a fix.

**Reply by ListechAu** [reply]
Over the Christmas - New Year break I was able to find some time to look into the slipstreaming a patch during installation issue I was having. While preparing a simple bundle to reproduce the issue (as @robmen suggested above) I believe I have identified the cause of the problem and it was nothing to do with the bundle at all.

In my package authoring I used: `<MediaTemplate EmbedCab="yes" />`

In my patch authoring I used: `<Media Id="1" Cabinet="$(FileName)_Patch.cab" EmbedCab="yes">`

The above code appears to result in a clash of DiskId in the Media table that can be seen in Orca when the MSI is opened and the MSP is loaded via the "Transform > View Patch" command. Only a single Cabinet entry is present, not one for the MSI and one for the MSP.

Simply changing the patch authoring to have a different "Id" (e.g. `<Media Id="2" Cabinet="$(FileName)_Patch.cab" EmbedCab="yes">`) seems to have fixed the uninstall problem I was having.

**Reply by robmen** [reply]
Might be a reasonable thing to open an issue to track for someone to potentially tackle in the patch build.

**Reply by ListechAu** [reply]
Thanks for the response Rob.
As suggested, I have created an "Issue" report for this.
